### PR TITLE
Support target selector

### PR DESCRIPTION
**Pull request by mdesmet**
### Problem

Fixes https://github.com/AltimateAI/vscode-dbt-power-user/issues/1106

### Solution

Describe the implemented solution. Add external references if needed.

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test

- Steps to be followed to verify the solution or code changes
- Mention if there is any settings configuration added/changed/deleted

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit c49463e25784ca94297fb2c587907387a9ee44bf  | 
|--------|--------|

feat: add target selector for dbt projects

### Summary:
This PR adds a target selector feature to dbt projects, enhancing target management and UI integration, and fixes issue #1106.

**Key points**:
- Fixes issue #1106 by adding a target selector feature to dbt projects.
- Introduces `get_target_names()` and `set_selected_target()` in `DbtProject`.
- Adds `dbtPowerUser.openTargetSelector` command for UI interaction.
- Implements `TargetStatusBar` for displaying and switching targets.
- Updates `DBTProjectIntegration` interface with new methods.
- Implements target selection in `dbtCoreIntegration.ts` and `dbtCloudIntegration.ts`.
- Renames `target` to `target_name` in `DbtProject` constructor.
- Updates `StatusBars` to include `TargetStatusBar`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

----

## Release Notes

- **New Features**
  - Introduced a target selection command in the VSCode interface, allowing users to easily choose from available targets.
  - Added a new `TargetStatusBar` component that displays the currently selected target in the status bar, enhancing visibility and interaction.
  - Enhanced target management functionality with new methods for setting and retrieving target names across various integration classes.

- **Bug Fixes**
  - Improved logic for updating target configurations to prevent overwriting existing values.

- **Documentation**
  - Minor formatting improvements for better code readability across various files.

### CODE REVIEW

File: src/statusbar/targetStatusBar.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      this.statusBar.hide();
+      this.dbtTerminal.debug(
+        "TargetStatusBar",
+        "Unable to update target status bar",
+        err,
+      );
+    }
+  }
+
+  private getCurrentProject() {
```
Comment (AdiGajbhiye) [inline src/statusbar/targetStatusBar.ts:67]: i think @saravmajestic created a helper util to get current project in `src/service/queryManifestService.ts`

**Review by AdiGajbhiye — APPROVED** [review]
lgtm

File: src/statusbar/targetStatusBar.ts (hunk 1 above)
Comment (mdesmet) [reply src/statusbar/targetStatusBar.ts:67]: it is not exactly the same. This picks the single project even if no text editor is open.

Comment (mdesmet) [reply src/statusbar/targetStatusBar.ts:67]: We should probably also move this logic to this helper service though.

**Comment by anandgupta42** [comment]
@mdesmet couple of questions:

1. What will be the behavior of the target status bar when "dbt is not detected"?
2. If targets are updated, will the file watcher re-initialize the target names?
3. In the error messages related to the target, will the user know if the target was picked from the picker and not from the 

Some suggestions:
1. we should add a logging event when target is swtiched
2. add a telemetry event to track the feature's popularity.
3. Not sure, how big target name can be, should we limit the length and show "..." ?

**Pull request merged**